### ISSUE TITLE
Accessability Violation : The ARIA attributes "aria-label" are not valid for the element <label> with ARIA role "none"

### ISSUE THREAD

**Issue by gqdeng** [carbon-design-system/carbon-components-angular#2950]
## Detailed description

> Describe in detail the issue you're having.

IBM Accessibility Checker tool reported that "The ARIA attributes "aria-label" are not valid for the element with ARIA role "none"" for the checkbox label tag of the table component selection column

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

Checkbox for table selection column

> What did you expect to happen? What happened instead? What would you like to see changed?

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

"carbon-components": "^11.48.1",
"carbon-components-angular": "^5.27.3",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

Run the IBM Accessibility Checker tool on table component with selection columns

## Additional information

* Screenshots or code

<img width="1718" alt="Screenshot 2024-07-30 at 11 37 42" src="https://github.com/user-attachments/assets/dcfc11a9-857f-45aa-9561-be93152d80a2">

<img width="597" alt="Screenshot 2024-07-30 at 16 41 56" src="https://github.com/user-attachments/assets/73d5e58c-a161-4b80-a1d4-ee118d542105">

**Comment by gqdeng** [comment]
the stackblitz used to repordice this issue:
[src code](https://stackblitz.com/edit/stackblitz-starters-mfmvdw?file=package.json,src%2Fmain.ts)
[preview page](https://stackblitzstartersmfmvdw-zwng--4200--41692973.local-credentialless.webcontainer.io/)

open the preview page above & run IBM Accessibility Checker tool on it can produce this issue.

<img width="1722" alt="Screenshot 2024-07-31 at 13 54 48" src="https://github.com/user-attachments/assets/40ec269f-2ba4-4d83-8e98-01104a3445f7">

**Comment by gqdeng** [comment]
Maybe remove the attr.aria-label from the label can fix this,  because the label element is inherently accessible and does not need an ARIA role or aria-label attributes.

[code to be removed]( https://github.com/carbon-design-system/carbon-components-angular/blob/master/src/checkbox/checkbox.component.ts#L54)

Please let me know if this works or not. Thanks
